### PR TITLE
X12 EDI Connector Reference Improvements

### PR DESCRIPTION
**Pull request by dhirsch1**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by sms-nvargas — DISMISSED** [review]

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 1]
```diff
@@ -8,77 +8,221 @@
...
+|Name | String | Name for this configuration. Connectors reference the configuration with this name. | |x
+| Expiration Policy a| ExpirationPolicy |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances when appropriate. |  |
+| Mule ID qualifier a| String a|  ID qualifier for Mule in the exchange (ISA) segment. This value is used in combination with the Self Identification Interchange ID value, and if one is set, the other must also be set. 
+{sp}+
+{sp}+
+If this value is set, it affects both send and receive message handling:
+
+* On the receive side, the ISA Interchange Receiver ID Qualifier must match this value. 
+* On the send side, this value is used as the default ISA Interchange Sender ID Qualifier when not set in the message data. |  |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* On the send side, this value is used as the default ISA Interchange Sender ID Qualifier when it is not set in the message data. |  |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 2]
```diff
@@ -8,77 +8,221 @@
...
+
+* On the receive side, the GS Application Receiver's Code must match this value. 
+* On the send side, this value is used as the default GS Application Sender's Code when it is not set in the message data. |  |
+| Partner ID qualifier a| String a|  ID qualifier for the partner in the exchange (ISA segment). This value is used in combination with the Partner Identification Interchange ID value, and if one is set the other must also be set.
+{sp}+
+{sp}+
+If this value is set, it affects both send and receive message handling:
+
+* On the receive side, the ISA Interchange Sender ID Qualifier must match this value. 
+* On the send side, this value is used as the default ISA Interchange Receiver ID Qualifier when this is not set in the message data. |  |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* On the send side, this value is used as the default ISA Interchange Receiver ID Qualifier when it is not set in the message data. |  |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 3]
```diff
@@ -8,77 +8,221 @@
...
+| Partner interchange ID a| String a| ID for the partner in the exchange (ISA segment). This value is used in combination with the Partner Identification Interchange ID qualifier value, and if one is set the other must also be set. 
+{sp}+
+{sp}+
+If this value is set, it affects both send and receive message handling:
+
+* On the receive side, the ISA Interchange Sender ID must match this value. 
+* On the send side, this value is used as the default ISA Interchange Receiver ID when this is not set in the message data. |  |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* On the send side, this value is used as the default ISA Interchange Receiver ID when it is not set in the message data. |  |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 4]
```diff
@@ -8,77 +8,221 @@
...
+| Partner application code a| String a| Application Code for the partner in the group (GS segment).
+{sp}+
+{sp}+
+If this value is set, it affects both send and receive message handling:
+
+* On the receive side, the GS Application Sender's Code must match this value. 
+* On the send side, this value is used as the default GS Application Receiver's Code when this is not set in the message data. |  |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* On the send side, this value is used as the default GS Application Receiver's Code when it is not set in the message data. |  |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 5]
```diff
@@ -8,77 +8,221 @@
...
 | Form and validation a| Enumeration, one of:
 
 ** X12
 ** HIPAA_SNIP1
-** HIPAA_SNIP2 | Schema form and validation level. Schemas used in the configuration must match the form selected, either X12 or HIPAA. SNIP Type 1 and Type 2 validation only applies to HIPAA schemas. |  X12 |
-| Schema definitions a| Array of String | List of paths for schemas to be used by connector. The paths may be either for file system or classpath. |  |x
+** HIPAA_SNIP2 | Schema form and validation level. Schemas used in the configuration must match the selected form. SNIP Type 1 and SNIP Type 2 validation applies to HIPAA schemas only. |  X12 |
+| Schema definitions a| Array of String | List of paths for the schemas that the connector uses.  The paths can be file systems or classpaths. |  |x
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Schema definitions a| Array of String | List of paths for the schemas that the connector uses. The paths can be file systems or classpaths. |  |x
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 6]
```diff
@@ -8,77 +8,221 @@
...
+| Allow unused segments a| Boolean a|  Whether to allow segments marked as `Unused` in a transaction set:
+
+* If `true`, the connector ignores the unused segments and does not reject the transactions. 
+* If `false`, the connector rejects transactions that contain segments marked as unused in the schema.
+
+In either case, the error is reported in 997/999 acknowledgments. |  false |
+| Enforce segment repeats a| Boolean a|  Whether to enforce segment repetition count limits in transaction sets. 
+
+* If `true`, the connector rejects transactions that repeat the segment too many times.
+* If false, the connector ignores the repeated segments and does not reject the transactions.
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* If `false`, the connector ignores the repeated segments and does not reject the transactions. 
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 7]
```diff
@@ -8,77 +8,221 @@
...
+* If `false`, you must directly specify the 997 schema, 999 schema, or both if you want these to be processed as input. The schemas used for generating 997 or 999 Functional Acknowledgments are hardcoded and cannot be modified. |  true |
 | Data element separator a| String |  Data element separator character. The configured value is used by default for all output messages, but can be overridden at the message level. |  * |
 | Component element separator a| String |  Component element separator character. The configured value is used by default for all output messages, but can be overridden at the message level. |  > |
-| Repetition separator a| String |  Repetition separator character (if used). The configured value is used by default for all output messages, but can be overridden at the message level. The value 'U' means repetitions are not used. |  U |
+| Repetition separator a| String |  Repetition separator character (if used). The configured value is used by default for all output messages, but can be overridden at the message level. The value `U`` means repetitions are not used. |  U |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Repetition separator a| String |  Repetition separator character (if used). The configured value is used by default for all output messages, but can be overridden at the message level. The value `U` means repetitions are not used. |  U |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 8]
```diff
@@ -8,77 +8,221 @@
...
+| Unique group numbers a| Boolean a|  Whether to send globally unique Group Control Numbers (GS06):
+
+* If `true`, the connector assigns unique group numbers across all interchanges sent to the same partner and application (as determined by the interchange sender and receiver identification, combined with the functional group sender and receiver application codes).
+* If `false`, the connector assigns functional group control numbers sequentially within each interchange and reuses the numbers in different interchanges.  |  false |
+| Unique transaction numbers a| Boolean a|  Whether to send globally unique Transaction Set Control Numbers (ST02): 
+
+* If `true`, the connector assigns unique transaction set numbers across all interchanges sent to the same partner and application (as determined by the interchange sender and receiver identification, combined with the functional group sender and receiver application codes).
+* If `false`, the connector assigns transaction set control numbers sequentially within each functional group and resuses the control numbers in different groups.
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* If `false`, the connector assigns transaction set control numbers sequentially within each functional group and reuses the control numbers in different groups. 
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 9]
```diff
@@ -8,77 +8,221 @@
...
+* If `false`, the connector assigns transaction set control numbers sequentially within each functional group and resuses the control numbers in different groups. 
+ |  false |
+| Implementation reference a| String |  Implementation convention reference for transactions (ST03). If this value is set, it is used as the ST03 Implementation Convention Reference unless it is overridden in the message parameters. |  |
+| Initial interchange number a| Number |  Initial Interchange Control Number used for outgoing messages. |  1 |
+| Initial group number a| Number |  Initial Group Control Number used for outgoing messages. |  1 |
+| Initial transaction number a| Number |  Initial Transaction Set Control Number used for outgoing messages. |  1 |
+| Requests acknowledgments a| Boolean a|  Whether to request acknowledgments such as TA1 for the sent transactions flag (ISA14):
+
+* If `true`, TA1 acknowledgments are requested for all sent transactions.
+* If `false`, TAI acknowledgements are not requested.|  false |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* If `false`, TA1 acknowledgements are not requested.|  false |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 10]
```diff
@@ -92,17 +236,17 @@ Enforce minimum and maximum lengths for Writer values. If true, a transaction wi
...
 
 [%header%autowidth.spread]
 |===
 | Name | Type | Description | Default Value | Required
-| Configuration | String | The name of the configuration to use. | |x
-| Read Content a| Binary |  |  `#[payload]` |
-| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
-| Target Value a| String |  An expression to evaluate against the operation's output and to store the outcome of that expression in the target variable. |  `#[payload]` |
+| Configuration | String | Name of the configuration to use. | |x
+| Read Content a| Binary | Input payload |  `#[payload]` |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Read Content a| Binary | Input payload. |  `#[payload]` |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 11]
```diff
@@ -128,21 +272,25 @@ Transform an input stream of EDI text into the maps and lists structure of the E
...
 | Name | Type | Description | Default Value | Required
-| Configuration | String | The name of the configuration to use. | |x
-| Write Content a| Object |  |  `#[payload]` |
-| Streaming Strategy a| * repeatable-in-memory-stream
-* repeatable-file-store-stream
-* non-repeatable-stream |  Configure if repeatable streams should be used and their behavior. |  |
-| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
-| Target Value a| String |  An expression to evaluate against the operation's output and to store the outcome of that expression in the target variable. |  `#[payload]` |
+| Configuration | String | Name of the configuration to use. | |x
+| Write Content a| Object | Output payload |  `#[payload]` |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Write Content a| Object | Output payload. |  `#[payload]` |
```

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 12]
```diff
@@ -167,17 +315,17 @@ Transform the maps and lists structure of EDI data into an EDI text stream.
...
 
 [%header%autowidth.spread]
 |===
 | Name | Type | Description | Default Value | Required
-| Configuration | String | The name of the configuration to use. | |x
-| Write Content a| Object |  |  `#[payload]` |
-| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
-| Target Value a| String |  An expression to evaluate against the operation's output and to store the outcome of that expression in the target variable. |  `#[payload]` |
+| Configuration | String | Name of the configuration to use. | |x
+| Write Content a| Object | Output payload |  `#[payload]` |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Write Content a| Object | Output payload. |  `#[payload]` |
```

**Review by sathya0 — DISMISSED** [review]

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 13]
```diff
@@ -128,21 +272,25 @@ Transform an input stream of EDI text into the maps and lists structure of the E
...
+| Streaming Strategy a| Enumeration, one of:
+
+* <<repeatable-in-memory-stream,Repeatable In Memory Stream>>
+* <<repeatable-file-store-stream,Repeatable File Store Stream>>
+* non-repeatable-stream
+| Configures how Mule processes streams. 
+|  | Repeatable file store stream |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]: remove Repeatable file store stream box as this table is not rendering correctly

File: x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 14]
```diff
@@ -8,77 +8,221 @@
...
+
+* On the receive side, the GS Application Receiver's Code must match this value. 
+* On the send side, this value is used as the default GS Application Sender's Code when it is not set in the message data. |  |
+| Partner ID qualifier a| String a|  ID qualifier for the partner in the exchange (ISA segment). This value is used in combination with the Partner Identification Interchange ID value, and if one is set the other must also be set.
+{sp}+
+{sp}+
+If this value is set, it affects both send and receive message handling:
+
+* On the receive side, the ISA Interchange Sender ID Qualifier must match this value. 
+* On the send side, this value is used as the default ISA Interchange Receiver ID Qualifier it is not set in the message data. |  |
```
Comment (sathya0) [inline x12-edi/2.9/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
* On the send side, this value is used as the default ISA Interchange Receiver ID Qualifier when it is not set in the message data. |  |
```

**Review by sathya0 — APPROVED** [review]

**Pull request merged**